src/internal/modules/contact-entry/forms/select-contact.js

```diff
@@ -0,0 +1,53 @@
+const { formFactory, fields } = require('shared/lib/forms');
+const Joi = require('@hapi/joi');
+const { uniqBy } = require('lodash');
+
+const getContactChoices = contacts => {
+  const choices = [...uniqBy(contacts, entity => [entity.id].join()).map(contact => ({
+    value: contact.id,
+    label: contact.name
+  })), { divider: 'or' }];
+  return [...choices.length > 1 ? choices : [], {
+    value: 'new',
+    label: 'Set up a new contact'
+  }];
+};
+
+const form = (request, defaultValue) => {
+  const { csrfToken } = request.view;
+  const { sessionKey, searchQuery, back, regionId } = request.query;
+  const { contactSearchResults } = request.pre;
+
+  const f = formFactory('/contact-entry/select-contact');
+
+  f.fields.push(fields.radio('id', {
+    errors: {
+      'any.required': {
+        message: 'Select a contact from the list'
```

Comment:
The above error message does not cover the 'set up new contact' option - it change to 'Select an existing contact or select set up a new contact' and if there are no existing contacts listed then it should just be 'Select set up a new contact'